offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -152,42 +159,81 @@ class OfferToPDFConverter implements OfferExporter {
-        def elementId = "product-items"+"-"+tableNum
-        // Create the items in html in the overview table
-        offer.items.each { item ->
+        def elementId = "product-items" + "-" + tableNum
 
-            if (itemPos % maxTableItems == 0) //start (next) table
-            {
-                elementId = "product-items"+"-"+ ++tableNum
-                htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
-            }
-            htmlContent.getElementById(elementId)
-                    .append(ItemPrintout.itemInHTML(itemPos++, item))
+        List<ProductItem> listOverheadItems = offer.itemsWithOverhead
+        List<ProductItem> listNoOverheadItems = offer.itemsWithoutOverhead
 
+        // Create the items in html in the overview table
+        if (listOverheadItems.size() > 0) {
+            listOverheadItems.each { productItem ->
+                if (currTableItems >= maxTableItems) //start (next) table
+                {
+                    println(productItem.product.productName)
+                    elementId = "product-items" + "-" + ++tableNum
+                    htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
+                    currTableItems = 1
+                }
+                htmlContent.getElementById(elementId)
+                        .append(ItemPrintout.itemInHTML(itemPos++, productItem))
+                currTableItems++
+            }
+            // Add subtotal Footer to ProductItem
+            htmlContent.getElementById(elementId).append(ItemPrintout.subTotalFooter(true))
+            // Include spacing for Footer
+            currTableItems = currTableItems + 2
+            // Reset ItemPosition
+            itemPos = 1
+            //Update Pricing of Footer
+            setIntermediatePrices(true)
+        }
+        if (listNoOverheadItems.size() > 0) {
+            listNoOverheadItems.each { productItem ->
+                if (currTableItems >= maxTableItems) //start (next) table
+                {
+                    println(productItem.product.productName)
+                    elementId = "product-items" + "-" + ++tableNum
+                    htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
+                    currTableItems = 1
+                }
+                htmlContent.getElementById(elementId)
+                        .append(ItemPrintout.itemInHTML(itemPos++, productItem))
+                currTableItems++
+            }
+            // Add subtotal Footer to ProductItem
+            htmlContent.getElementById(elementId).append(ItemPrintout.subTotalFooter(false))
+            // Reset ItemPosition
+            itemPos = 1
+            // Include Spacing for Footer
+            currTableItems = currTableItems + 2
+            //Update Pricing of Footer
+            setIntermediatePrices(false)
+        }
+        if (currTableItems >= maxTableItems - 4) {
```

Comment:
why -4 ?

Reply:
Originally it was to prevent overflowing the table through the addition of the total cost. With the method adaption this is not necessary anymore